pkg/sippyserver/dbloader.go

```diff
@@ -0,0 +1,125 @@
+package sippyserver
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+	"k8s.io/klog"
+
+	"github.com/openshift/sippy/pkg/buganalysis"
+	"github.com/openshift/sippy/pkg/db"
+	"github.com/openshift/sippy/pkg/db/models"
+	"github.com/openshift/sippy/pkg/testgridanalysis/testgridconversion"
+	"github.com/openshift/sippy/pkg/testgridanalysis/testidentification"
+	"github.com/openshift/sippy/pkg/testgridanalysis/testreportconversion"
+)
+
+func (a TestReportGeneratorConfig) PrepareDatabase(
+	dbc *db.DB,
+	dashboard TestGridDashboardCoordinates,
+	variantManager testidentification.VariantManager,
+	syntheticTestManager testgridconversion.SyntheticTestManager,
+	bugCache buganalysis.BugCache, // required to associate tests with bug
+) error {
+	testGridJobDetails, _ := a.TestGridLoadingConfig.load(dashboard.TestGridDashboardNames)
+	rawJobResultOptions := testgridconversion.ProcessingOptions{
+		SyntheticTestManager: syntheticTestManager,
+		// Load the last 30 days of data.  Note that we do not prune data today, so we'll be accumulating
+		// data over time for now.
+		StartDay: 0,
+		NumDays:  30,
+	}
+	rawJobResults, _ := rawJobResultOptions.ProcessTestGridDataIntoRawJobResults(testGridJobDetails)
+
+	// allJobResults holds all the job results with all the test results.  It contains complete frequency information and
+	allJobResults := testreportconversion.ConvertRawJobResultsToProcessedJobResults(
+		dashboard.ReportName, rawJobResults, bugCache, dashboard.BugzillaRelease, variantManager)
+
+	// Load all job and test results into database:
+	klog.Info("loading ProwJobs into db")
+
+	// Build up a cache of all prow jobs we know about to speedup data entry.
+	// Maps job name to db ID.
+	prowJobCache := map[string]uint{}
+	var idNames []models.IDName
+	dbc.DB.Model(&models.ProwJob{}).Find(&idNames)
```

Comment:
Can we make sure this is filterable?

Reply:
Looks like i'll need to redo once we move to the query, will address in followup.